### DISCUSSION TITLE
How can velero backup trigger CloudNativePG backup?

### DISCUSSION THREAD

**Discussion by retrogaming457** [vmware-tanzu/velero#7746] [Community support Q&A]
Sometimes we need to backup an application as well as it's backend database at the same time. For example, for a application using postgres as its backend, I want to backup application's namespace using velero and postgres database using backup facilities of postgres itself (e.g. CloudNativePG operator) - not velero. 

I understand that both Velero and CloudNativePG have cronjob and we can set them both for the same time to occur. But this is somehow repeating approach which we normally want to avoid. So my question is:

How can we trigger CloundNativePG backup when velero starts backing up state of the application?

**Comment by sseago** [comment]
You might be able to do this by executing the CloudNativePG backup in a backup hook for the postgres pod:
https://velero.io/docs/v1.13/backup-hooks/

**Comment by djeinstine** [comment]
Hi @retrogaming457 , did you figure something out in the end?

I understand I could simply trigger the `barman-cloud-backup` using hooks. However, what about restore? I haven't seen a reasonable way to do it. According to cloudnativepg, you need to add externalClusters to the spec to a fresh pg cluster. However, as I understand it, velero would simply apply the pg cluster spec as it was stored on k8s.

Is it really as simple as triggering a `barman-cloud-wal-restore` on a velero restore hook?

**Reply by dragoangel** [reply]
I think better approach would be spinning up new cluster in another namespace from `volumeSnapshot` and that's it

**Comment by gcezaralmeida** [comment]
@retrogaming457 @djeinstine, Hey Guys, I'm interested on how you guys solved this issue?
